light-rpc/src/main.rs

```diff
@@ -1,4 +1,503 @@
+
+        let confirmation_slot = rpc_client
+            .get_slot_with_commitment(CommitmentConfig {
+                commitment: CommitmentLevel::Confirmed,
+            })
+            .unwrap();
+        let finalized_slot = rpc_client
+            .get_slot_with_commitment(CommitmentConfig {
+                commitment: CommitmentLevel::Finalized,
+            })
+            .unwrap();
+
+        let confirmation_slot = Arc::new(AtomicU64::new(confirmation_slot));
+        let finalized_slot = Arc::new(AtomicU64::new(finalized_slot));
+
+        // start thread which will listen to confirmed blocks
+        {
+            let signature_status = signature_status.clone();
+            let confirmation_slot = confirmation_slot.clone();
+            let finalized_slot = finalized_slot.clone();
+            let join_handle = Builder::new()
+                .name("thread working on confirmation block".to_string())
+                .spawn(move || {
+                    Self::process_block(
+                        receiver_confirmed,
+                        signature_status,
+                        CommitmentLevel::Confirmed,
+                        confirmation_slot,
+                    );
+                })
+                .unwrap();
+            joinables.push(join_handle);
+        }
+        // start thread which will listen to finalized blocks
+        {
+            let signature_status = signature_status.clone();
+            let confirmation_slot = confirmation_slot.clone();
+            let finalized_slot = finalized_slot.clone();
+            let join_handle = Builder::new()
+                .name("thread working on finalized block".to_string())
+                .spawn(move || {
+                    Self::process_block(
+                        receiver_finalized,
+                        signature_status,
+                        CommitmentLevel::Finalized,
+                        finalized_slot,
+                    );
+                })
+                .unwrap();
+            joinables.push(join_handle);
+        }
+
+        LightRpcRequestProcessor {
+            rpc_client,
+            tpu_client,
+            last_valid_block_height: 0,
+            ws_url: websocket_url.clone(),
+            current_slot,
+            signature_status,
+            connection_cache: connection_cache,
+            subscribed_clients: Arc::new(vec![client_confirmed, client_finalized]),
+            joinables: Arc::new(joinables),
+            confirmation_slot: confirmation_slot,
+            finalized_slot: finalized_slot,
+        }
+    }
+
+    fn process_block(
+        reciever: Receiver<RpcResponse<RpcBlockUpdate>>,
+        signature_status: Arc<RwLock<HashMap<String, Option<CommitmentLevel>>>>,
+        commitment: CommitmentLevel,
+        updated_slot: Arc<AtomicU64>,
+    ) {
+        let mut last_update_slot: u64 = 0;
+        loop {
+            let block_data = reciever.recv();
+
+            match block_data {
```

Comment:
suggest to refactor towards the happy-path-should-be-left-aligned style (see https://maelvls.dev/go-happy-line-of-sight/)

Reply:
Is this possible in rust?